daemon/cmd/daemon_main.go

```diff
@@ -1768,31 +1700,6 @@ func runDaemon() {
 	}
 	bootstrapStats.k8sInit.End(true)
 	restoreComplete := d.initRestore(restoredEndpoints)
-	if wgAgent != nil {
-		if err := wgAgent.RestoreFinished(); err != nil {
-			log.WithError(err).Error("Failed to set up wireguard peers")
-		}
-	}
-
-	if d.endpointManager.HostEndpointExists() {
-		d.endpointManager.InitHostEndpointLabels(d.ctx)
-	} else {
```

Comment:
I think there's actually a bug in today's code, this should have a hard dependency on `<-d.initRestore(...)`. Otherwise this "host endpoint exists" check will always return false, because `initRestore()` is the one that's responsible for restoring endpoints from disk and then calling the `endpointManager.Expose()` function to expose the endpoint to the rest of the system. If `initRestore()` is not yet complete, then there may actually be a host endpoint that is stored on disk, but not yet exposed in the EndpointManager. So we'd restore that one and then also proceed to the `else` case here and add it again.

cc @pchaigno